activerecord/lib/active_record/persistence.rb

```diff
@@ -1256,7 +1256,7 @@ def _create_record(attribute_names = self.attribute_names)
       )
 
       returning_columns.zip(returning_values).each do |column, value|
-        _write_attribute(column, value) if !_read_attribute(column)
+        _write_attribute(column, value)
```

Comment:
MySQL, as always, "surprises" by its behaviour. 

Rails uses `SHOW FULL FILEDS` to get the table's schema info. This works for views too. But, for regular tables for primary keys it returns something like
```
mysql> SHOW FULL FIELDS FROM books;
+------------------------+--------------+--------------------+------+-----+---------+----------------+---------------------------------+---------+
| Field                  | Type         | Collation          | Null | Key | Default | Extra          | Privileges                      | Comment |
+------------------------+--------------+--------------------+------+-----+---------+----------------+---------------------------------+---------+
| id                     | int(11)      | NULL               | NO   | PRI | NULL    | auto_increment | select,insert,update,references |         |
``` 

but for views it returns something like this
```
mysql> SHOW FULL FIELDS FROM printed_books;
+--------+--------------+--------------------+------+-----+---------+-------+---------------------------------+---------+
| Field  | Type         | Collation          | Null | Key | Default | Extra | Privileges                      | Comment |
+--------+--------------+--------------------+------+-----+---------+-------+---------------------------------+---------+
| id     | int(11)      | NULL               | NO   |     | 0       |       | select,insert,update,references |         |
```

(note the that the default is set to 0 😱)

So we need to remove this check, because it will always be set to 0. Not sure why it was added initially, seems like redundant.

Reply:
Ok, CI fails, turns out that this is needed, because MySQL does not support `RETURNING`.

MySQL databases have autoincrement keys starting from 1, but it is possible to configure them to start from 0. How is it safe to assume that real-world applications never do this? With this assumption I can make this code work for MariaDB.

Reply:
This is a great find, we end up merging https://github.com/rails/rails/pull/50783 which fixes views for postgresql but keeps mysql views broken so we still need to look into it